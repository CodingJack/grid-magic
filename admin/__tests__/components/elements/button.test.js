/**
 * External dependencies.
 */
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

/**
 * Internal dependencies.
 */
import Button from '../../../src/js/components/buttons/button';

const namespace = 'esg-v4';
Enzyme.configure( { adapter: new Adapter() } );

const text = 'This is a button';

test( 'it renders correctly', () => {
  const tree = create(
    <Button 
      namespace={ namespace } 
      onClick={ () => {} } 
    /> 
  ).toJSON();
  
  if( ! tree ) {
    return;
  }
  expect( tree ).toMatchSnapshot();
  const { props } = tree;

  props.icon = 'check';
  props.onClick = () => {};
  props.disabled = true;
  props.large = true;
  props.namespace = 'name-space';
  props.color = "green";
  props.text = text;
  props.extraClass = `.${ namespace }-decorator`;
  props.children = 'some text';

  expect( tree ).toMatchSnapshot();
} );

describe('onClick callback works and it unmounts', () => {
  let clicked;
  const component = shallow(
    <Button 
      namespace={ namespace } 
      text={ text } 
      icon="check" 
      onClick={ () => clicked = true } 
    /> 
  );

  it('should match the snapshot', () => {
    expect( component.html() ).toMatchSnapshot();
  });

  test( 'Click event', () => {
    component.simulate( 'click', {} );
    expect( clicked ).toEqual( true );
  } );

  test( 'it unmounts', () => {
    component.unmount();
  } );
} );




