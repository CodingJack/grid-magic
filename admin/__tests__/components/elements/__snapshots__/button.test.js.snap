// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="gmagic-error-js"
>
  <p>
    Something went wrong.  Please 
    <a
      href="https://github.com/codingjack/gmagic/issues"
    >
      report
    </a>
     the error below.
    <i>
      A Button Component requires text or an icon
    </i>
  </p>
  <p>
    <i>
      A Button Component requires text or an icon
    </i>
  </p>
</div>
`;

exports[`it renders correctly 2`] = `
<div
  children="some text"
  className="gmagic-error-js"
  color="green"
  disabled={true}
  extraClass=".gmagic-decorator"
  icon="check"
  large={true}
  namespace="name-space"
  onClick={[Function]}
  text="This is a button"
>
  <p>
    Something went wrong.  Please 
    <a
      href="https://github.com/codingjack/gmagic/issues"
    >
      report
    </a>
     the error below.
    <i>
      A Button Component requires text or an icon
    </i>
  </p>
  <p>
    <i>
      A Button Component requires text or an icon
    </i>
  </p>
</div>
`;

exports[`onClick callback works and it unmounts should match the snapshot 1`] = `"<button class=\\"gmagic-btn \\" role=\\"button\\" tabindex=\\"0\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z\\"></path></svg><span>This is a button</span></button>"`;
